lib/ansible/plugins/strategy/__init__.py

```diff
@@ -866,7 +866,8 @@ def _evaluate_conditional(h):
         elif meta_action == 'clear_facts':
             if _evaluate_conditional(target_host):
                 for host in self._inventory.get_hosts(iterator._play.hosts):
-                    self._variable_manager.clear_facts(host)
+                    hostname = host.get_name()
+                    self._variable_manager.clear_facts(hostname)
```

Comment:
Fine by me :) I just modelled this after 3101e24 as I assumed it was already following best practice.